FileHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using HwCreateGame;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+
+namespace HwCreateGame
+{
+    public class FileHandler
+    {
+        private string filePath;
+        private Mutex mutex;
+
+        public FileHandler(string filePath)
+        {
+            this.filePath = filePath;
+            mutex = new Mutex();
+        }
+
+        // Запись контактов в файл
+        public void WriteContactsToFile(List<Contact> contacts)
+        {
+            mutex.WaitOne();
```

Comment:
The task was not to lock with a mutex but to ensure only the first instance may write to the contact list (including an internal collection of contacts). Here you use mutex as lock, which is 1: unnecessary, cause you can handle that with the File class 2: wrong by itself